### ISSUE TITLE
 How can I make multiple "prebuild_config" for different schemas?

### ISSUE THREAD

**Issue by mefilt** [grab/cocoapods-binary-cache#18]
How can I make multiple "prebuild_config" for different schemas?

**Comment by trinhngocthuyen** [comment]
Hi @mefilt, Could you please help describe your use case in more detail?

**Comment by mefilt** [comment]
There are many configurations in my project.
And I need to have a cache for each configuration
How can I do this?
<img width="837" alt="image" src="https://user-images.githubusercontent.com/1247906/92105725-f020cb80-edeb-11ea-957b-14f3d32caada.png">

**Comment by mefilt** [comment]
I came up with this option, but it doesn't work either, since the cache in the repository is also shared
`plugin "cocoapods-binary-cache"
if ENV["CI_ENABLE"].nil?
  
  config_cocoapods_binary_cache(
      prebuild_config: "Dev-Debug",
      dev_pods_enabled: false,
      prebuild_all_vendor_pods: true
    )
else

  config_cocoapods_binary_cache(
      prebuild_config: "Release-Prod",
      dev_pods_enabled: false,
      prebuild_all_vendor_pods: true
    )
end
`
Is it possible to do it automatically? And save caches in different folders?

**Comment by trinhngocthuyen** [comment]
@mefilt Thanks for the clarification. You are correct about the fact that the cache repo is currently shared and cannot support multiple configurations.

Moving forward, I would like to propose a **multi-cache-repo support** to solve this problem as follows:

- The configuration file (`PodBinaryCacheConfig.json`) would look like this:
```rb
{
  "prebuild_path": "_Prebuild",
  "cache_repos": {
    "default": {
      "remote": "<url/to/cache/repo>",
      "local": "<path/to/cache/dir>"
    },
    "another": { # 👈  <--- Use this repo for another configuration
      "remote": "<url/to/cache/repo>",
      "local": "<path/to/cache/dir>"
    }
  }
}
```
- When running the CLI, you can specify the repo (if not specified, use `default`). For example:
```sh
bundle exec pod binary fetch --repo=another
```
- The `prebuild_config` will stay in the `Podfile`. You can alternate it by `ENV` like you mentioned
```rb
config_cocoapods_binary_cache(
  prebuild_config: ENV["CI"] ? "Test" : "Debug"
)
```

In the meantime, you can try out a workaround by having a script to dynamically generate the `PodBinaryCacheConfig.json` file.

**Comment by mefilt** [comment]
Oooyyee! Thx man. 
Json looks confy.

**Comment by mefilt** [comment]
You can add a "remote url" field to "config_cocoapods_binary_cache".

And you still need to store binaries for the simulator and devices in the repository at the same time

**Comment by trinhngocthuyen** [comment]
**1/ About the cache for simulators & devices**

Currently, we do support prebuilt frameworks for devices. However, we don't store cache for simulators & devices separately. Rather, the binary of a prebuilt framework is a fat binary with architectures used for both simulator and devices.
Kindly refer to the `device_build_enabled` option in [this doc](https://github.com/grab/cocoapods-binary-cache/blob/master/docs/configure_cocoapods_binary_cache.md) to enable device support.

**2/ About the suggestion to have the `remote_url` option in `config_cocoapods_binary_cache`**

That would be best.

Originally, we had those cache-repo-related configurations in the json file instead of `config_cocoapods_binary_cache` to _make the cache fetch/push step independent of Podfile_ (which requires loading the Podfile). With this given use case and more similar cases in the future, we probably consider migrating these options to `config_cocoapods_binary_cache` as suggested. We'll discuss this approach among the team. Thanks for the suggestion!

**Comment by trinhngocthuyen** [comment]
@mefilt The support has been implemented in [0.1.7](https://github.com/grab/cocoapods-binary-cache/releases/tag/0.1.7).

Kindly refer to the **`cache_repo`** option in this doc for the configuration: https://github.com/grab/cocoapods-binary-cache/blob/0.1.7/docs/configure_cocoapods_binary_cache.md

Thanks for reporting the issue and the suggestion.
